astropy/time/core.py

```diff
@@ -408,6 +409,40 @@ def now(cls):
 
     info = TimeInfo()
 
+    @classmethod
+    def strptime(cls, time_string, format_string):
+        """
+        Parse a string to a Time according to a format specification.
+        See time.strptime documentation for format specification.
+
+        >>> Time.strptime('2012-06-30T23:59:60', '%Y-%m-%dT%H:%M:%S')
+        <Time object: scale='utc' format='isot' value=2012-06-30T23:59:60.000>
+
+        Parameters
+        ----------
+        time_string : string, sequence, ndarray
+            Objects containing time data of type string
+        format_string : string
+            String specifying format of time_string.
+
+        Returns
+        -------
+        astropy.time.Time
+            A new `Time` object corresponding to the input time_string.
+
+        """
+        time_array = np.array(time_string)
+        to_string = (str if time_array.dtype.kind == 'U' else
```

Comment:
Check that `val.dtype.kind in ('U', 's')`, and if not raise a `TypeError` exception that the input needs to be strings.  Add a corresponding test of that exception being raised for bad input.